### PR TITLE
Adding environment variables required by logger in worker

### PR DESCRIPTION
**Pull request by gauravgahlot**
## Description

The PR adds environment variables required by `packethost/pkg/log` in the worker. 

## Why is this needed

Required for the dependent PR [#tink-271](https://github.com/tinkerbell/tink/pull/271).

## How Has This Been Tested?

Tested the changes by executing an end-to-end workflow. 

## Worker Logs
```
{"log":"{\"level\":\"info\",\"ts\":1598612543.5217416,\"caller\":\"tink-worker/main.go:48\",\"msg\":\"starting version: 6960d5f\",\"service\":\"github.com/tinkerbell/tink\"}\r\n","stream":"stdout","time":"2020-08-28T11:02:23.526199655Z"}
{"log":"{\"level\":\"info\",\"ts\":1598612543.5223653,\"caller\":\"tink-worker/main.go:85\",\"msg\":\"RETRY_INTERVAL not set, using default 3 seconds\",\"service\":\"github.com/tinkerbell/tink\"}\r\n","stream":"stdout","time":"2020-08-28T11:02:23.526229686Z"}
{"log":"{\"level\":\"info\",\"ts\":1598612543.5223973,\"caller\":\"tink-worker/main.go:99\",\"msg\":\"MAX_RETRY not set, using default 3 retries\",\"service\":\"github.com/tinkerbell/tink\"}\r\n","stream":"stdout","time":"2020-08-28T11:02:23.526236012Z"}
{"log":"{\"level\":\"info\",\"ts\":1598612543.5452757,\"caller\":\"tink-worker/worker.go:142\",\"msg\":\"starting with action\",\"service\":\"github.com/tinkerbell/tink\",\"workerId\":\"0eba0bf8-3772-4b4a-ab9f-6ebe93b90a94\",\"workflowId\":\"b3a88595-5bfd-40ed-bd32-766adeab4fa2\",\"actionName\":\"hello_world\",\"taskName\":\"hello world\"}\r\n","stream":"stdout","time":"2020-08-28T11:02:23.545666793Z"}
{"log":"{\"level\":\"info\",\"ts\":1598612543.5510116,\"caller\":\"tink-worker/worker.go:166\",\"msg\":\"sent action status\",\"service\":\"github.com/tinkerbell/tink\",\"workerId\":\"0eba0bf8-3772-4b4a-ab9f-6ebe93b90a94\",\"workflowId\":\"b3a88595-5bfd-40ed-bd32-766adeab4fa2\",\"actionName\":\"hello_world\",\"taskName\":\"hello world\"}\r\n","stream":"stdout","time":"2020-08-28T11:02:23.551388082Z"}
{"log":"{\"status\":\"Pulling from hello-world\",\"id\":\"latest\"}\r\r\n","stream":"stdout","time":"2020-08-28T11:02:23.644575307Z"}
{"log":"{\"status\":\"Pulling fs layer\",\"progressDetail\":{},\"id\":\"0e03bdcc26d7\"}\r\r\n","stream":"stdout","time":"2020-08-28T11:02:23.644659223Z"}
{"log":"{\"status\":\"Downloading\",\"progressDetail\":{\"current\":730,\"total\":2529},\"progress\":\"[==============\\u003e                                    ]     730B/2.529kB\",\"id\":\"0e03bdcc26d7\"}\r\r\n","stream":"stdout","time":"2020-08-28T11:02:23.66942451Z"}
{"log":"{\"status\":\"Downloading\",\"progressDetail\":{\"current\":2529,\"total\":2529},\"progress\":\"[==================================================\\u003e]  2.529kB/2.529kB\",\"id\":\"0e03bdcc26d7\"}\r\r\n","stream":"stdout","time":"2020-08-28T11:02:23.669436361Z"}
{"log":"{\"status\":\"Verifying Checksum\",\"progressDetail\":{},\"id\":\"0e03bdcc26d7\"}\r\r\n","stream":"stdout","time":"2020-08-28T11:02:23.669439988Z"}
{"log":"{\"status\":\"Download complete\",\"progressDetail\":{},\"id\":\"0e03bdcc26d7\"}\r\r\n","stream":"stdout","time":"2020-08-28T11:02:23.66944226Z"}
{"log":"{\"status\":\"Extracting\",\"progressDetail\":{\"current\":2529,\"total\":2529},\"progress\":\"[==================================================\\u003e]  2.529kB/2.529kB\",\"id\":\"0e03bdcc26d7\"}\r\r\n","stream":"stdout","time":"2020-08-28T11:02:23.66944429Z"}
{"log":"{\"status\":\"Extracting\",\"progressDetail\":{\"current\":2529,\"total\":2529},\"progress\":\"[==================================================\\u003e]  2.529kB/2.529kB\",\"id\":\"0e03bdcc26d7\"}\r\r\n","stream":"stdout","time":"2020-08-28T11:02:23.66944685Z"}
{"log":"{\"status\":\"Pull complete\",\"progressDetail\":{},\"id\":\"0e03bdcc26d7\"}\r\r\n","stream":"stdout","time":"2020-08-28T11:02:23.731208642Z"}
{"log":"{\"status\":\"Digest: sha256:90659bf80b44ce6be8234e6ff90a1ac34acbeb826903b02cfa0da11c82cbc042\"}\r\r\n","stream":"stdout","time":"2020-08-28T11:02:23.731223756Z"}
{"log":"{\"status\":\"Status: Downloaded newer image for 192.168.1.1/hello-world:latest\"}\r\r\n","stream":"stdout","time":"2020-08-28T11:02:23.731226604Z"}
{"log":"{\"level\":\"info\",\"ts\":1598612543.6948822,\"caller\":\"tink-worker/action.go:203\",\"msg\":\"starting container with cmd: []\",\"service\":\"github.com/tinkerbell/tink\",\"workflowId\":\"b3a88595-5bfd-40ed-bd32-766adeab4fa2\",\"workerId\":\"0eba0bf8-3772-4b4a-ab9f-6ebe93b90a94\",\"actionName\":\"hello_world\"}\r\n","stream":"stdout","time":"2020-08-28T11:02:23.731228638Z"}
{"log":"\r\n","stream":"stdout","time":"2020-08-28T11:02:23.864270483Z"}
{"log":"Hello from Docker!\r\n","stream":"stdout","time":"2020-08-28T11:02:23.864278931Z"}
{"log":"This message shows that your installation appears to be working correctly.\r\n","stream":"stdout","time":"2020-08-28T11:02:23.864281213Z"}
{"log":"\r\n","stream":"stdout","time":"2020-08-28T11:02:23.864283136Z"}
{"log":"To generate this message, Docker took the following steps:\r\n","stream":"stdout","time":"2020-08-28T11:02:23.864284705Z"}
{"log":" 1. The Docker client contacted the Docker daemon.\r\n","stream":"stdout","time":"2020-08-28T11:02:23.864286483Z"}
{"log":" 2. The Docker daemon pulled the \"hello-world\" image from the Docker Hub.\r\n","stream":"stdout","time":"2020-08-28T11:02:23.864288907Z"}
{"log":"    (amd64)\r\n","stream":"stdout","time":"2020-08-28T11:02:23.864290903Z"}
{"log":" 3. The Docker daemon created a new container from that image which runs the\r\n","stream":"stdout","time":"2020-08-28T11:02:23.864292477Z"}
{"log":"    executable that produces the output you are currently reading.\r\n","stream":"stdout","time":"2020-08-28T11:02:23.864294192Z"}
{"log":" 4. The Docker daemon streamed that output to the Docker client, which sent it\r\n","stream":"stdout","time":"2020-08-28T11:02:23.864295866Z"}
{"log":"    to your terminal.\r\n","stream":"stdout","time":"2020-08-28T11:02:23.864297598Z"}
{"log":"\r\n","stream":"stdout","time":"2020-08-28T11:02:23.864299188Z"}
{"log":"To try something more ambitious, you can run an Ubuntu container with:\r\n","stream":"stdout","time":"2020-08-28T11:02:23.864300687Z"}
{"log":" $ docker run -it ubuntu bash\r\n","stream":"stdout","time":"2020-08-28T11:02:23.864302378Z"}
{"log":"\r\n","stream":"stdout","time":"2020-08-28T11:02:23.864303968Z"}
{"log":"Share images, automate workflows, and more with a free Docker ID:\r\n","stream":"stdout","time":"2020-08-28T11:02:23.86430546Z"}
{"log":" https://hub.docker.com/\r\n","stream":"stdout","time":"2020-08-28T11:02:23.864307212Z"}
{"log":"\r\n","stream":"stdout","time":"2020-08-28T11:02:23.864308793Z"}
{"log":"For more examples and ideas, visit:\r\n","stream":"stdout","time":"2020-08-28T11:02:23.86431029Z"}
{"log":" https://docs.docker.com/get-started/\r\n","stream":"stdout","time":"2020-08-28T11:02:23.864311973Z"}
{"log":"\r\n","stream":"stdout","time":"2020-08-28T11:02:23.864313659Z"}
{"log":"{\"level\":\"info\",\"ts\":1598612543.9096792,\"caller\":\"tink-worker/action.go:266\",\"msg\":\"starting to remove container: eff322d7a5a7bada3d179be6ec927ac2fcac8873c677d2d91d79cfe3fee359f2\",\"service\":\"github.com/tinkerbell/tink\",\"workflowId\":\"b3a88595-5bfd-40ed-bd32-766adeab4fa2\",\"workerId\":\"0eba0bf8-3772-4b4a-ab9f-6ebe93b90a94\",\"actionName\":\"hello_world\"}\r\n","stream":"stdout","time":"2020-08-28T11:02:23.911821295Z"}
{"log":"{\"level\":\"info\",\"ts\":1598612543.9113615,\"caller\":\"tink-worker/action.go:86\",\"msg\":\"container removed with status: ACTION_SUCCESS\",\"service\":\"github.com/tinkerbell/tink\",\"workflowId\":\"b3a88595-5bfd-40ed-bd32-766adeab4fa2\",\"workerId\":\"0eba0bf8-3772-4b4a-ab9f-6ebe93b90a94\",\"actionName\":\"hello_world\"}\r\n","stream":"stdout","time":"2020-08-28T11:02:23.911833548Z"}
{"log":"{\"level\":\"info\",\"ts\":1598612543.9113853,\"caller\":\"tink-worker/action.go:133\",\"msg\":\"action container exit status code: ACTION_SUCCESS\",\"service\":\"github.com/tinkerbell/tink\",\"workflowId\":\"b3a88595-5bfd-40ed-bd32-766adeab4fa2\",\"workerId\":\"0eba0bf8-3772-4b4a-ab9f-6ebe93b90a94\",\"actionName\":\"hello_world\"}\r\n","stream":"stdout","time":"2020-08-28T11:02:23.91183715Z"}
{"log":"{\"level\":\"info\",\"ts\":1598612543.9146202,\"caller\":\"tink-worker/worker.go:211\",\"msg\":\"sent action status\",\"service\":\"github.com/tinkerbell/tink\",\"workerId\":\"0eba0bf8-3772-4b4a-ab9f-6ebe93b90a94\",\"workflowId\":\"b3a88595-5bfd-40ed-bd32-766adeab4fa2\",\"actionName\":\"hello_world\",\"taskName\":\"hello world\"}\r\n","stream":"stdout","time":"2020-08-28T11:02:23.915028437Z"}
{"log":"{\"level\":\"info\",\"ts\":1598612543.916867,\"caller\":\"tink-worker/worker.go:217\",\"msg\":\"reached to end of workflow\",\"service\":\"github.com/tinkerbell/tink\",\"workerId\":\"0eba0bf8-3772-4b4a-ab9f-6ebe93b90a94\",\"workflowId\":\"b3a88595-5bfd-40ed-bd32-766adeab4fa2\"}\r\n","stream":"stdout","time":"2020-08-28T11:02:23.917199393Z"}
```

### CODE REVIEW

**Comment by gianarb** [comment]
This is not related to this pr but I think the logging library we use is weird

```
{"log":"{"level":"info","ts":1598596063.0076628,"caller":"tink-worker/main.go:85","msg":"RETRY_INTERVAL not set, using default 3 seconds","service":"github.com/tinkerbell/tink"}\r\n","stream":"stdout","time":"2020-08-28T06:27:43.008612455Z"}
```
This line can not be parsed by Go for example and it is not a valid JSON.

https://play.golang.com/p/wQm7xpcKRBq

The string inside `log` is a valid JSON. What am I doing wrong? Are we using that library internally?

```
{"log":"To try something more ambitious, you can run an Ubuntu container with:\r\n","stream":"stdout","time":"2020-08-28T06:27:43.633684029Z"}
```

This is not even a JSON.

The structure of this output looks very unstructured. @mmlb do you have any context around this? I think we should just onboard something a bit more traditional like `zap` and probably we can have some common interface if needed. But I think I am missing something big

**Comment by gauravgahlot** [comment]
> This is not related to this pr but I think the logging library we use is weird
> 
> ```
> {"log":"{"level":"info","ts":1598596063.0076628,"caller":"tink-worker/main.go:85","msg":"RETRY_INTERVAL not set, using default 3 seconds","service":"github.com/tinkerbell/tink"}\r\n","stream":"stdout","time":"2020-08-28T06:27:43.008612455Z"}
> ```
> 
> This line can not be parsed by Go for example and it is not a valid JSON.
> 
> https://play.golang.com/p/wQm7xpcKRBq
> 
> The string inside `log` is a valid JSON. What am I doing wrong? Are we using that library internally?
> 
> ```
> {"log":"To try something more ambitious, you can run an Ubuntu container with:\r\n","stream":"stdout","time":"2020-08-28T06:27:43.633684029Z"}
> ```
> 
> This is not even a JSON.
> 
> The structure of this output looks very unstructured. @mmlb do you have any context around this? I think we should just onboard something a bit more traditional like `zap` and probably we can have some common interface if needed. But I think I am missing something big

Apologies. The attached log output was somehow messed up. 
I got the logs again and ran it through your code. The Go code is now able to parse the log lines, so it's a valid JSON.

[https://play.golang.com/p/a1qkFuKwGAF](https://play.golang.com/p/a1qkFuKwGAF)

**Comment by gianarb** [comment]
There are still messages that are JSON and others that are not, and that's weird to me

File: apps/workflow-helper.sh [hunk 1]
```diff
@@ -71,6 +71,10 @@ docker run --privileged -t --name "tink-worker" \
 	-e "TINKERBELL_CERT_URL=$grpc_cert_url" \
 	-e "REGISTRY_USERNAME=$registry_username" \
 	-e "REGISTRY_PASSWORD=$registry_password" \
+	-e "ROLLBAR_TOKEN=ignored" \
+	-e "ROLLBAR_DISABLE=-1" \
```
Comment (mmlb) [inline apps/workflow-helper.sh:75]: These were made optional in a recent version of the lib.

**Comment by mmlb** [comment]
There should be nothing weird here, its just a wrapper around zap with some "as best used at packet" sprinkling on top. Every message going through the log library should be valid json, any other messages are likely sent straight to stdout via fmt.Println or similar. Got an example?

**Review by mmlb — CHANGES_REQUESTED** [review]

**Comment by gianarb** [comment]
@mmlb the examples are the pull request itself. Some line has JSON as the content of the message, other has a string.

```
{"log":"{\"status\":\"Pulling fs layer\",\"progressDetail\":{},\"id\":\"0e03bdcc26d7\"}\r\r\n","stream":"stdout","time":"2020-08-28T11:02:23.644659223Z"}
```

```
{"log":" 4. The Docker daemon streamed that output to the Docker client, which sent it\r\n","stream":"stdout","time":"2020-08-28T11:02:23.864295866Z"}
```

File: apps/workflow-helper.sh (hunk 1 above)
Comment (gauravgahlot) [reply apps/workflow-helper.sh:75]: Sorry, I don't follow. The logger still panics if the token is not set. 
https://github.com/packethost/pkg/blob/master/log/internal/rollbar/rollbar.go#L21

Am I overlooking something here?

Comment (mmlb) [reply apps/workflow-helper.sh:75]: Oops I thought this was merged already. https://github.com/packethost/pkg/pull/31. Let me check on it and lets hold and wait for it.

Comment (gauravgahlot) [reply apps/workflow-helper.sh:75]: This looks like an old PR. Is it in shape so we can move it forward?

**Comment by mmlb** [comment]
> @mmlb the examples are the pull request itself. Some line has JSON as the content of the message, other has a string.

I think the pr might be incomplete, I say so because:
These **do** look like messages generated from github.com/packethost/pkg/log
```
{"log":"{\"level\":\"info\",\"ts\":1598612543.5452757,\"caller\":\"tink-worker/worker.go:142\",\"msg\":\"starting with action\",\"service\":\"github.com/tinkerbell/tink\",\"workerId\":\"0eba0bf8-3772-4b4a-ab9f-6ebe93b90a94\",\"workflowId\":\"b3a88595-5bfd-40ed-bd32-766adeab4fa2\",\"actionName\":\"hello_world\",\"taskName\":\"hello world\"}\r\n","stream":"stdout","time":"2020-08-28T11:02:23.545666793Z"}
{"log":"{\"level\":\"info\",\"ts\":1598612543.5510116,\"caller\":\"tink-worker/worker.go:166\",\"msg\":\"sent action status\",\"service\":\"github.com/tinkerbell/tink\",\"workerId\":\"0eba0bf8-3772-4b4a-ab9f-6ebe93b90a94\",\"workflowId\":\"b3a88595-5bfd-40ed-bd32-766adeab4fa2\",\"actionName\":\"hello_world\",\"taskName\":\"hello world\"}\r\n","stream":"stdout","time":"2020-08-28T11:02:23.551388082Z"}
```
the log field has a json with level, ts, caller, msg, service keys, these are always generated.

But these lines **do not** look like github.com/packethost/pkg/log messages:
```
{"log":"{\"status\":\"Pulling from hello-world\",\"id\":\"latest\"}\r\r\n","stream":"stdout","time":"2020-08-28T11:02:23.644575307Z"}
{"log":"{\"status\":\"Pulling fs layer\",\"progressDetail\":{},\"id\":\"0e03bdcc26d7\"}\r\r\n","stream":"stdout","time":"2020-08-28T11:02:23.644659223Z"}
```
These actually look like the output of `docker exec/run -t` maybe.

**Review by gianarb — APPROVED** [review]

File: apps/workflow-helper.sh (hunk 1 above)
Comment (mmlb) [reply apps/workflow-helper.sh:75]: @gauravgahlot https://github.com/packethost/pkg/pull/31 has been merged if you update go mod and pull it in we can avoid messing with env vars.

**Review by mmlb — CHANGES_REQUESTED** [review]

**Comment by gauravgahlot** [comment]
Since [packethost/pkg#31](https://github.com/packethost/pkg/pull/31) has been merged, we do not need to pass any additional environment variables to the worker. 
Therefore, I'm closing this PR.

**Pull request closed without merging**